torchaudio/functional.py

```diff
@@ -1239,6 +1240,60 @@ def dcshift(
     return output_waveform
 
 
+def overdrive(
+        waveform: Tensor,
+        gain: float = 20,
+        colour: float = 20
+) -> Tensor:
+    r"""Apply a overdrive effect to the audio. Similar to SoX implementation.
+    This effect applies a non linear distortion to the audio signal.
+
+    Args:
+        waveform (Tensor): audio waveform of dimension of `(..., time)`
+        gain (float): desired gain at the boost (or attenuation) in dB
+            Allowed range of values are 0 to 100
+        colour (float):  controls the amount of even harmonic content in the over-driven output
+            Allowed range of values are 0 to 100
+
+    Returns:
+        Tensor: Waveform of dimension of `(..., time)`
+
+    References:
+        http://sox.sourceforge.net/sox.html
+    """
+    actual_shape = waveform.shape
+
+    if len(actual_shape) == 2:
+        waveform = waveform.unsqueeze(0)
+
+    gain = _dB2Linear(gain)
+    colour = colour / 200
+    last_in = torch.zeros(waveform.shape[0], waveform.shape[1])
+    last_out = torch.zeros(waveform.shape[0], waveform.shape[1])
+
+    temp = waveform * gain + colour
+
+    mask1 = temp < -1
+    temp[mask1] = temp[mask1] * 0 - 2 / 3
+    # Above redundant multiplcation to accomadate torchscript issue
+
+    mask2 = temp > 1
+    temp[mask2] = temp[mask2] * 0 + 2 / 3
+
+    mask3 = (~mask1 & ~mask2)
+    temp[mask3] = temp[mask3] - (temp[mask3]**3) * (1. / 3)
+
+    output_waveform = torch.zeros_like(waveform)
+
+    # TODO: Implement a torch CPP extension
```

Comment:
Out of curiosity, do you plan to do this? (I am not asking you to do so.)

Reply:
yeah . haven't written any cpp extension till now , but will try to do it sometime later . The python version is very slow